src/lib.rs

```diff
@@ -115,6 +115,31 @@ where
     fs::copy(from.as_ref(), to.as_ref()).map_err(|source| CopyError::new(source, from, to))
 }
 
+/// Wrapper for [`fs::remove_dir`](https://doc.rust-lang.org/stable/std/fs/fn.remove_dir.html).
+pub fn remove_dir<P>(path: P) -> io::Result<()>
+where
+    P: AsRef<Path> + Into<PathBuf>,
+{
+    fs::remove_dir(path.as_ref()).map_err(|source| Error::new(source, ErrorKind::RemoveDir, path))
+}
+
+/// Wrapper for [`fs::remove_dir_all`](https://doc.rust-lang.org/stable/std/fs/fn.remove_dir_all.html).
+pub fn remove_dir_all<P>(path: P) -> io::Result<()>
+where
+    P: AsRef<Path> + Into<PathBuf>,
+{
+    fs::remove_dir_all(path.as_ref())
+        .map_err(|source| Error::new(source, ErrorKind::RemoveDir, path))
```

Comment:
Do you think we need to pull the directory traversal into this crate? If we invoke `fs::remove_dir_all`, then if there's a permission error removing a deeply-nested file, fs-err will falsely report the permission error as stemming from the root path.

For API coverage purposes, it might be enough to start with this approach.

Reply:
Just had a go at this. Unfortunately the standard library implementation on windows uses the unstable [`FileTypeExt::is_symlink_dir`](https://doc.rust-lang.org/nightly/std/os/windows/fs/trait.FileTypeExt.html#tymethod.is_symlink_dir).